src/main/java/net/dries007/tfc/objects/items/metal/ItemMetalBucket.java

```diff
@@ -0,0 +1,190 @@
+                        }
+                        if (!playerIn.isCreative())
+                        {
+                            // In creative, buckets function but don't give new items
+                            ItemHandlerHelper.giveItemToPlayer(playerIn, result.getResult());
+                        }
+                        return ActionResult.newResult(EnumActionResult.SUCCESS, stack);
+                    }
+                }
+                else
+                {
+                    //Filled bucket, try to empty it.
+                    IBlockState stateAt = worldIn.getBlockState(pos);
+                    if (!(stateAt.getBlock().isReplaceable(worldIn, pos) || !stateAt.getMaterial().isSolid()))
+                    {
+                        //if cant place in the block we found, try the side of it
+                        pos = pos.offset(rayTrace.sideHit);
+                    }
+                    stateAt = worldIn.getBlockState(pos);
+                    if (stateAt.getBlock().isReplaceable(worldIn, pos) || !stateAt.getMaterial().isSolid())
+                    {
+                        if (!stateAt.getMaterial().isLiquid())
+                        {
+                            worldIn.destroyBlock(pos, true); //drop the replaceable block (ie: torches, flowers...)
+                        }
+
+                        FluidStack fluidStack = bucketCap.drain(CAPACITY, true);
+                        if (fluidStack != null)
+                        {
+                            Fluid fluid = fluidStack.getFluid();
+                            // Place a flowing water block, not a source
+                            if (fluid.getBlock() instanceof BlockFluidBase)
+                            {
+                                worldIn.setBlockState(pos, fluid.getBlock().getDefaultState());
+                            }
+                            else if (fluid.getBlock() instanceof BlockLiquid)
+                            {
+                                // Vanilla water and lava, and hopefully nothing else
+                                try
+                                {
+                                    BlockLiquid flowingBlock = BlockStaticLiquid.getFlowingBlock(fluid.getBlock().getDefaultState().getMaterial());
+                                    worldIn.setBlockState(pos, flowingBlock.getDefaultState());
+                                }
+                                catch (IllegalArgumentException e) { /* Just skip */ }
+                            }
+                        }
+                        worldIn.playSound(null, pos, SoundEvents.ITEM_BUCKET_EMPTY, SoundCategory.PLAYERS, 1.0F, 1.0F);
+                        return ActionResult.newResult(EnumActionResult.SUCCESS, stack);
+                    }
+                }
+            }
+        }
+        return ActionResult.newResult(EnumActionResult.PASS, stack);
```

Comment:
Prefer just calling constructors (there's a couple other places where this can be changed as well)
```suggestion
        return new ActionResult<>(EnumActionResult.PASS, stack);
```

Reply:
This mimics code in ItemWoodenBucket. Happy to change in both places, if that's what we want?

Reply:
Yes